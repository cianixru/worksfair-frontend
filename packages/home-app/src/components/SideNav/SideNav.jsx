import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { resolve } from 'path';
import { NavLink } from 'react-router-dom';
import keygen from 'keygenerator';

class SideNav extends Component {
  dashNav = React.createRef()

  toggleDashNav = (event) => {
    event.preventDefault();

    const dashNav = this.dashNav.current;
    dashNav.classList.toggle('active-nav');
  }

  render() {
    const {
      links: {
        main, webpage, settings,
      }
    } = this.props;
    return (
      <div>
        <div className="dashboard-button">
          <button
            className="button is-medium is-fullwidth is-dark"
            onClick={this.toggleDashNav}>
            <span className="icon">
              <i className="fa fa-navicon" />
            </span>
            <span>Dashboard Navigation</span>
          </button>
        </div>
        <div
          className="dashboard-nav"
          ref={this.dashNav}
          // style={{ display }}
        >
          <div className="dashboard-nav-inner">
            <ul data-submenu-title="Main">
              {main.map(link => (
                <li
                  key={keygen.transaction_id()}
                  className={
                    window.location.pathname === link.to ? 'active' : null
                  }
                >
                  <NavLink to={resolve(link.to)} activeClassName="active">
                    <i className={link.icon} aria-hidden="true" />
                    {link.text}
                  </NavLink>
                </li>
              ))}
            </ul>

            <ul data-submenu-title="Webpages">
              {webpage.map(link => (
                <li
                  key={keygen.transaction_id()}
                  className={
                    window.location.pathname === link.to ? 'active' : null
                  }
                >
                  <NavLink to={resolve(link.to)} activeClassName="active">
                    <i className={link.icon} aria-hidden="true" />
                    {link.text}
                  </NavLink>
                </li>
              ))}
            </ul>

            <ul data-submenu-title="Settings">
              {settings.map(link => (
                <li
                  key={keygen.transaction_id()}
                  className={
                    window.location.pathname === link.to ? 'active' : null
                  }
                >
                  <NavLink to={resolve(link.to)} activeClassName="active">
                    <i className={link.icon} aria-hidden="true" />
                    {link.text}
                  </NavLink>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

SideNav.propTypes = {
  links: PropTypes.object,
};

export default SideNav;
