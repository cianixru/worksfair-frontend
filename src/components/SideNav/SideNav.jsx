import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { resolve } from 'path';
import { NavLink } from 'react-router-dom';
import keygen from 'keygenerator';

class SideNav extends Component {
  render() {
    const {
      links: {
        main, webpage, settings,
      }
    } = this.props;
    return (
      <div className="dashboard-nav">
        <div className="dashboard-nav-inner">
          <ul data-submenu-title="Main">
            {main.map(link => (
              <li
                key={keygen.transaction_id()}
                className={
                  window.location.pathname === link.to ? 'active' : null
                }
              >
                <NavLink to={resolve(link.to)} activeClassName="active">
                  <i className={link.icon} aria-hidden="true" />
                  {link.text}
                </NavLink>
              </li>
            ))}
          </ul>

          <ul data-submenu-title="Webpages">
            {webpage.map(link => (
              <li
                key={keygen.transaction_id()}
                className={
                  window.location.pathname === link.to ? 'active' : null
                }
              >
                <NavLink to={resolve(link.to)} activeClassName="active">
                  <i className={link.icon} aria-hidden="true" />
                  {link.text}
                </NavLink>
              </li>
            ))}
          </ul>

          <ul data-submenu-title="Settings">
            {settings.map(link => (
              <li
                key={keygen.transaction_id()}
                className={
                  window.location.pathname === link.to ? 'active' : null
                }
              >
                <NavLink to={resolve(link.to)} activeClassName="active">
                  <i className={link.icon} aria-hidden="true" />
                  {link.text}
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}

SideNav.propTypes = {
  links: PropTypes.object,
};

export default SideNav;
