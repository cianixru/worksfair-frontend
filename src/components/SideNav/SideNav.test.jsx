import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import SideNav from './SideNav';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({});

describe('DashboardHeader.jsx', () => {
  const links = {
    main: [
      {
        text: 'Dashboard',
        to: '/dashboard/theo',
      },
      {
        text: 'Create Webpage',
        to: '/dashboard/theo/webpage/new',
      },
    ],
    webpage: [
      {
        text: 'Create Webpage',
        to: '/dashboard/theo/webpage/new',
      },
    ],
    settings: [
      {
        text: 'Create Webpage',
        to: '/dashboard/theo/webpage/new',
      },
    ],
  };
  test('Should display the SideNave Component with links', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <SideNav
            links={links}
          />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
