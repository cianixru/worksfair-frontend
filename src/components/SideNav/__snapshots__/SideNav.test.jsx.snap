// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardHeader.jsx Should display the SideNave Component with links 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dashboard-nav"
      >
        <div
          class="dashboard-nav-inner"
        >
          <ul
            data-submenu-title="Main"
          >
            <li>
              <a
                href="/dashboard/theo"
              >
                <i
                  aria-hidden="true"
                />
                Dashboard
              </a>
            </li>
            <li>
              <a
                href="/dashboard/theo/webpage/new"
              >
                <i
                  aria-hidden="true"
                />
                Create Webpage
              </a>
            </li>
          </ul>
          <ul
            data-submenu-title="Webpages"
          >
            <li>
              <a
                href="/dashboard/theo/webpage/new"
              >
                <i
                  aria-hidden="true"
                />
                Create Webpage
              </a>
            </li>
          </ul>
          <ul
            data-submenu-title="Settings"
          >
            <li>
              <a
                href="/dashboard/theo/webpage/new"
              >
                <i
                  aria-hidden="true"
                />
                Create Webpage
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dashboard-nav"
    >
      <div
        class="dashboard-nav-inner"
      >
        <ul
          data-submenu-title="Main"
        >
          <li>
            <a
              href="/dashboard/theo"
            >
              <i
                aria-hidden="true"
              />
              Dashboard
            </a>
          </li>
          <li>
            <a
              href="/dashboard/theo/webpage/new"
            >
              <i
                aria-hidden="true"
              />
              Create Webpage
            </a>
          </li>
        </ul>
        <ul
          data-submenu-title="Webpages"
        >
          <li>
            <a
              href="/dashboard/theo/webpage/new"
            >
              <i
                aria-hidden="true"
              />
              Create Webpage
            </a>
          </li>
        </ul>
        <ul
          data-submenu-title="Settings"
        >
          <li>
            <a
              href="/dashboard/theo/webpage/new"
            >
              <i
                aria-hidden="true"
              />
              Create Webpage
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
